fiftyone/utils/csv.py

```diff
@@ -214,3 +219,246 @@ def _parse_value(value):
+                raise ValueError(
+                    f"Unsupported fields parameter of type {type(fields)}"
+                )
+            if any("." in f for f in fields):
+                raise ValueError(
+                    "Embedded fields not supported in fields parameter"
+                )
+
+        self.field_parsers = fields
+
+    def import_samples(self, dataset, tags=None):
+        """Imports samples and labels stored on disk in CSV format.
+
+        Args:
+            dataset: a :class:`fiftyone.core.dataset.Dataset`
+            tags (None): an optional list of tags to attach to each sample
+
+        Returns:
+            a list of IDs of the samples that were added to the dataset
+        """
+
+        with open(self.labels_path, "r") as labels_file:
+            labels_csv = csv.DictReader(labels_file)
+            if self.media_field not in labels_csv.fieldnames:
+                raise ValueError(
+                    f"No media field '{self.media_field}'"
+                    f" found in '{self.labels_path}'"
+                )
+
+            # Sane defaults
+            #   None: all columns as strings
+            #   List: All columns in list as strings
+            #   Dict: Column name -> Parser or function
+            if self.field_parsers is None:
+                _field_parsers = {
+                    field: Parsers.DefaultParser
+                    for field in labels_csv.fieldnames
+                }
+            elif isinstance(self.field_parsers, list):
+                _field_parsers = {
+                    field: Parsers.DefaultParser
+                    for field in self.field_parsers
+                }
+            elif isinstance(self.field_parsers, dict):
+                _field_parsers = {
+                    f: p or Parsers.DefaultParser
+                    for f, p in self.field_parsers.items()
+                }
+            else:
+                raise ValueError(
+                    f"Unsupported fields parameter of type {type(self.field_parsers)}"
+                )
+
+            embedded_fields = [
+                f.split(".")[0] for f in labels_csv.fieldnames if "." in f
+            ]
+            if len(embedded_fields) != len(set(embedded_fields)):
+                raise ValueError(
+                    "Cannot import embedded document type with multiple fields"
+                )
+
+            has_tags = self.TAGS_FIELD in labels_csv.fieldnames
+            samples = []
+            with fou.ProgressBar(total=self.max_samples) as pb:
+                # Rip through csv rows
+                for csv_row in pb(labels_csv):
+                    media_loc = csv_row.pop(self.media_field)
+                    # Not absolute path (either local or remote)
+                    if not os.path.isabs(media_loc) and not re.match(
+                        r"^\w+://", media_loc
+                    ):
+                        media_loc = os.path.join(self.data_path, media_loc)
```

Comment:
Is this right? Do we want to reconstitute a relative path to an absolute path, relative to `self.data_path`? What if there's no media included?